scripts/src/main/resources/scripts/functions

```diff
@@ -1134,6 +1135,13 @@ function doGitPullOrClone() {
       popd > /dev/null || exit 255 
     fi
   fi
+  
+  if [ "${url/.git/}" != "${url}" ]
+  then
+    doGitPullOrClone "${target_path}" "${url}"
```

Comment:
1. Variable `target_path` is undefined here. Did you mean `dir`instead?
2. This seems to cause an infinity loop.